fix set_() with functionalization

bdhirsh opened the pull request:
This should fix https://github.com/pytorch/pytorch/issues/90573

Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #91001
* #90791
* __->__ #90722

bdhirsh committed:
fix set_() with functionalization

[ghstack-poisoned]

bdhirsh committed:
Update on "fix set_() with functionalization"

This should fix https://github.com/pytorch/pytorch/issues/90573




[ghstack-poisoned]

Skylion007 commented on aten/src/ATen/templates/RegisterFunctionalization.cpp:
```diff
@@ -36,6 +37,17 @@ constexpr auto exclude_keys_for_meta_dispatch =
+        allocator->allocate(t.nbytes()),
+        allocator,
+        /*resizeable=*/t.resizable());
+    return c10::Storage(storage_impl);
```
```suggestion
    return c10::Storage(std::move(storage_impl));
```

bdhirsh committed:
Update on "fix set_() with functionalization"

This should fix https://github.com/pytorch/pytorch/issues/90573




[ghstack-poisoned]

bdhirsh committed:
Update on "fix set_() with functionalization"

This should fix https://github.com/pytorch/pytorch/issues/90573




[ghstack-poisoned]

bdhirsh committed:
Update on "fix set_() with functionalization"

This should fix https://github.com/pytorch/pytorch/issues/90573




[ghstack-poisoned]

bdhirsh committed:
Update on "fix set_() with functionalization"

This should fix https://github.com/pytorch/pytorch/issues/90573




[ghstack-poisoned]

ezyang reviewed (approved):
lol
